tokio/src/runtime/scheduler/current_thread.rs

```diff
@@ -377,6 +377,46 @@ impl Handle {
         handle
     }
 
+    /// Capture a snapshot of this runtime's state.
+    #[cfg(all(tokio_unstable, feature = "taskdump"))]
+    pub(crate) fn dump(&self) -> crate::runtime::Dump {
+        use crate::runtime::{dump, task::trace::Trace};
+
+        let mut snapshots = vec![];
+
+        // todo: how to make this work outside of a runtime context?
+        CURRENT.with(|maybe_context| {
+            // drain the local queue
+            let Some(context) = maybe_context else { return };
+            let mut maybe_core = context.core.borrow_mut();
+            let Some(core) = maybe_core.as_mut() else { return };
+            let local = &mut core.tasks;
+            let _ = local.drain(..);
+
+            // drain the injection queue
+            if let Some(injection) = self.shared.queue.lock().as_mut() {
+                let _ = injection.drain(..);
+            }
+
+            // notify each task
+            let mut tasks = vec![];
+            self.shared.owned.for_each(|task| {
+                // set the notified bit
+                let _ = task.as_raw().state().transition_to_notified_for_tracing();
+                // store the raw tasks into a vec
+                tasks.push(task.as_raw());
```

Comment:
I would probably encapsulate this, because I don't want ref-count logic to happen outside of `src/runtime/task`.

Reply:
Resolved by https://github.com/tokio-rs/tokio/pull/5608/commits/88ef47e962cb22cc967ce502f9371fc731dc745e?